### PR TITLE
Fixes #34298 - support KEEP_CHANGELOG_LIMIT option

### PR DESCRIPTION
**Pull request by wbclark**
This was added to Pulpcore with https://github.com/pulp/pulp_rpm/pull/2333

Backport to 3.14 is underway

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -216,6 +219,7 @@
   Array[Stdlib::Absolutepath] $allowed_import_path = ['/var/lib/pulp/sync_imports'],
   Array[Stdlib::Absolutepath] $allowed_export_path = [],
   Pulpcore::ChecksumTypes $allowed_content_checksums = ['sha224', 'sha256', 'sha384', 'sha512'],
+  Integer[1] $keep_changelog_limit = undef,
```
Comment (ekohl) [inline manifests/init.pp:222]: This is not a valid type. It should be
```suggestion
  Optional[Integer[1]] $keep_changelog_limit = undef,
```

That said, the feature is explicit in that `0` should be the lowest value. See https://github.com/pulp/pulp_rpm/commit/cbc5e98770da6469bed936070b4f1c07fb5835c8#

**Review by ekohl — CHANGES_REQUESTED** [review]
This feature belongs to the RPM plugin, so everything should be happening in [pulpcore::plugin::rpm](https://github.com/theforeman/puppet-pulpcore/blob/master/manifests/plugin/rpm.pp).

Here's an example where we set config for a plugin:
https://github.com/theforeman/puppet-pulpcore/blob/d347abdac8091ed725a532427c3aa380a63bcbc9/manifests/plugin/container.pp#L37

**Comment by wbclark** [comment]
I pushed a commit which addresses all feedback and moves the feature to pulpcore::plugin::rpm

Will squash it after running all tests

File: manifests/plugin/rpm.pp [hunk 2]
```diff
@@ -1,8 +1,12 @@
 # @summary Pulp RPM plugin
 # @param use_pulp2_content_route
 #   Whether to redirect the legacy (Pulp 2) URLs to the content server
+#
+# @param keep_changelog_limit
+#   Pulpcore's KEEP_CHANGELOG_LIMIT setting. Uses Pulpcore's default when undefined. Increasing this limit will cause pulpcore workers to use more memory when more changelogs are available in the repo metadata.
```
Comment (ekohl) [inline manifests/plugin/rpm.pp:6]: I usually limit the line length, like this:
https://github.com/theforeman/puppet-pulpcore/blob/d347abdac8091ed725a532427c3aa380a63bcbc9/manifests/init.pp#L30-L33

I use vim to do that (shift v -> gq wraps).

**Review by ekohl — COMMENTED** [review]
Very small nits, but this is indeed what I expected. Please address them and squash the commit.

One thing that we can't do anything about, but I wonder if the Pulp setting should have been prefixed so it's `RPM_KEEP_CHANGELOG_LIMIT`. Or will pulp_deb also respect it? Just thinking out loud now.

File: manifests/plugin/rpm.pp (hunk 2 above)
Comment (wbclark) [reply manifests/plugin/rpm.pp:6]: Hugely useful. Thanks!

**Review by ekohl — APPROVED** [review]
Thanks!

**Pull request merged**